### PR TITLE
Update chicoma-cpu gnu and nvidia builds

### PR DESCRIPTION
**Pull request by xylar**
Following the recent DST, this merge updates the module files and environment variables on Chicoma-CPU.  We note that these updates work well for `gnu` and `nvidia` compilers but not yet for `intel`, which we are continuing to work on.  A separate update will be needed to address Chicoma-GPU as well.

An environment variable has also been added to the MPAS standalone framework to accommodate overriding the library path for standard c++ on Chicoma.

See https://github.com/E3SM-Ocean-Discussion/E3SM/pull/112 for relevant discussion

[BFB] for all E3SM testing

### CODE REVIEW

**Comment by xylar** [comment]
I have successfully run:
```
SMS_D.TL319_oQU240wLI_ais8to30.MPAS_LISIO_JRA1p5.chicoma-cpu_gnu
SMS_D.TL319_oQU240wLI_ais8to30.MPAS_LISIO_JRA1p5.chicoma-cpu_nvidia
```

**Comment by xylar** [comment]
I'll rebase once that goes in.

File: components/mpas-framework/Makefile [hunk 1]
```diff
@@ -396,11 +396,11 @@ gnu-cray:
 	"FFLAGS_OPT = -O3 -m64 -ffree-line-length-none -fconvert=big-endian -ffree-form -ffpe-summary=none $${EXTRA_FFLAGS}" \
 	"CFLAGS_OPT = -O3 -m64" \
 	"CXXFLAGS_OPT = -O3 -m64" \
-	"LDFLAGS_OPT = -O3 -m64" \
+	"LDFLAGS_OPT = -O3 -m64 $(GNU_CRAY_LDFLAGS)" \
 	"FFLAGS_DEBUG = -g -m64 -ffree-line-length-none -fconvert=big-endian -ffree-form -fbounds-check -fbacktrace -ffpe-trap=invalid,zero,overflow -ffpe-summary=none $${EXTRA_FFLAGS}" \
 	"CFLAGS_DEBUG = -g -m64" \
 	"CXXFLAGS_DEBUG = -g -m64" \
-	"LDFLAGS_DEBUG = -g -m64" \
+	"LDFLAGS_DEBUG = -g -m64 $(GNU_CRAY_LDFLAGS)" \
```
Comment (xylar) [inline components/mpas-framework/Makefile:403]: @matthewhoffman and @mark-petersen, this is what I want your review on.  If you want to look at other changes or run tests, that would be appreciated but it's up to you.

**Review by matthewhoffman — APPROVED** [review]
@xylar , this solution for Chicoma looks good and seems reasonable given the current constraints.  Thanks for sorting through this issue.

**Review by mark-petersen — APPROVED** [review]
Thanks very much @xylar! I used the head of compass and tested this branch with gnu, both debug and optimized. It passes all tests for the nightly test suite for MPAS-Ocean standalone. 

I also tested E3SM on this branch with
```
cd cime/scripts
./create_test SMS_Ld3.T62_oQU240.GMPAS-NYF.chicoma-cpu_gnu \
  -p t24_coastal_ocean --walltime 00:30:00
```
and it passes all steps.

**Comment by xylar** [comment]
> I used the head of compass and tested this branch with gnu, both debug and optimized.

Thanks for testing, but how did that work?  I'm working on Compass support in https://github.com/MPAS-Dev/compass/pull/865 but I wouldn't have expected anyone to be able to build standalone MPAS-Ocean or create a working Compass enviornment on Chicoma until that goes in.  And that requires a mache update as well.

**Comment by jonbob** [comment]
Passes:
* ERP_Ld3.ne30pg2_r05_IcoswISC30E3r5.WCYCL1850.chrysalis_intel.allactive-pioroot1

and verified tests build and run on chicoma-cpu using gnu and nvidia

Merged to next

**Comment by xylar** [comment]
Thanks @jonbob!

**Pull request merged**